src/codegen/compile.ml

```diff
@@ -5779,10 +6015,13 @@ module Stabilization = struct
         StableMem.write_word32 env ^^
 
         (* copy data to following stable memory *)
-        compile_const_64 4L ^^
-        extend64 get_dst ^^
-        extend64 get_len ^^
-        IC.system_call env "stable64_write"
+        Externalization.Strm.finalize_buffer
+          begin
+            compile_const_64 4L ^^
+            extend64 get_dst ^^
+            extend64 get_len ^^
+            IC.system_call env "stable64_write"
+          end
```

Comment:
```suggestion
        Externalization.Strm.finalize_buffer
          begin
            compile_const_64 4L ^^
            extend64 get_dst ^^
            extend64 get_len ^^
            IC.system_call env "stable64_write"
          end
```
Is it possible to just refactor this to:
```
 Externalization.Strm.finalize_buffer get_dst get_len
```
and hid the details in the implementation of finalize_buffer?

Reply:
I consider that as a layering violation. If `BumpStream` had a `finalize_buffer`, how should that be implemented? I cannot do `stable64_write`.
/me scratches head

Reply:
(the nop implementaton would just pop the stack)